raft/server.cc

```diff
@@ -987,8 +987,8 @@ future<> server_impl::applier_fiber() {
         while (true) {
             auto v = co_await _apply_entries.pop_eventually();
 
-            if (std::holds_alternative<std::vector<log_entry_ptr>>(v)) {
-                auto& batch = std::get<0>(v);
+            if (auto* batch_opt = std::get_if<std::vector<log_entry_ptr>>(&v)) {
```

Comment:
I thought we agree to move to `visitor`, no? As of `get_if` I do not mind either way, but, as previously discussed, I do not see what it improves (I counted the same amount of branches, and I certainly do not think it is more elegant) and if we will refactor each peace of code we touch to our preferred ways we will have a lot of useless back and forth patches. If there is no clear benefit just use the way the established code is already using.

Reply:
> I thought we agree to move to visitor, no? 

You said you don't mind, but I'm not sure about the visitor, it might introduce some additional allocations (coroutine inside visitor lambda), so I think `get_if` option is safer.

> but, as previously discussed, I do not see what it improves (I counted the same amount of branches,

I think the `x != null` comparison is cheaper than the check `std::get` must make (but TBH I'm not sure).
Also the code is more obvious to me now, compared to `std::get<0>` (the `0`... meh)

I can remove the patch and use the old way if the change pains you too much.

Reply:
For a new code I do not mind how we access a variant, but I do mind when a code that a developer touches is rewritten for its own liking just because he needs to touch it. `visitor` improves the situation because it lets you know it you missed a case, but your change is just cosmetic.